library/forms/lib/fieldRegistry/fields/TakePhoto/index.tsx

```diff
@@ -0,0 +1,794 @@
+ */
+const TakePhotoFull: React.FC<FullTakePhotoFieldProps> = props => {
+  const {
+    label,
+    helperText,
+    required,
+    advancedHelperText,
+    disabled = false,
+    state,
+    setFieldAttachment,
+    context,
+  } = props;
+
+  const [noPermission, setNoPermission] = useState(false);
+  const [hasLoadErrors, setHasLoadErrors] = useState(false);
+
+  // Get attachment service (guaranteed to exist in full mode)
+  const attachmentService = context.attachmentEngine();
+
+  // Get current photos from field state
+  const photos: StoredPhoto[] = (state.value?.attachments || []).map(att => ({
+    attachmentId: att.attachmentId,
+  }));
+
+  /**
+   * Captures a photo from the device camera with geolocation on native platforms
+   */
+  const takePhoto = useCallback(async () => {
+    try {
+      const isWeb = Capacitor.getPlatform() === 'web';
+
+      // Check/request camera permission
+      if (isWeb) {
+        const permission = await navigator.permissions.query({
+          name: 'camera' as PermissionName,
+        });
+        if (permission.state === 'denied') {
+          setNoPermission(true);
+          return;
+        }
+      } else {
+        const permissions = await Camera.requestPermissions({
+          permissions: ['camera'],
+        });
+        if (permissions.camera === 'denied') {
+          setNoPermission(true);
+          return;
+        }
+      }
+
+      // Capture photo
+      const photoResult = await Camera.getPhoto({
+        quality: 90,
+        allowEditing: false,
+        resultType: isWeb ? CameraResultType.Base64 : CameraResultType.Uri,
+        correctOrientation: true,
+        promptLabelHeader: 'Take or select a photo',
+      });
+
+      let photoBlob: Blob;
+
+      if (isWeb) {
+        photoBlob = await base64ImageToBlob(photoResult);
+      } else {
+        // Native: add geolocation EXIF data if possible
+        try {
+          const position = await Geolocation.getCurrentPosition({
+            enableHighAccuracy: true,
+            timeout: 10000,
+            maximumAge: 0,
+          });
+
+          if (position && photoResult.path) {
+            await Exif.setCoordinates({
+              pathToImage: photoResult.path,
+              lat: position.coords.latitude,
+              lng: position.coords.longitude,
+            });
+          }
+        } catch (e) {
+          console.warn('Could not add geolocation to photo:', e);
+        }
+
+        // Fetch the photo as a blob
+        if (!photoResult.webPath) {
+          throw new Error('Photo webPath is undefined');
+        }
+        const response = await fetch(photoResult.webPath);
+        photoBlob = await response.blob();
+      }
+
+      // Store the attachment
+      const timestamp = new Date().toISOString();
+      const filename = `photo_${timestamp}.${photoResult.format}`;
+
+      const result = await attachmentService.storeAttachmentFromBlob({
```

Comment:
Could you write this at the FormManager level and pass in a function to be called here rather than requiring the field to do this complex work? Then here we could just call props.storeAttachment() and might not need to pass in record Id etc.